### PR TITLE
feat(4-pdp): Add PDP route, page, cubit, api

### PR DESCRIPTION
**Pull request by stephenrichter**
This PR adds the basic PDP feature functionality.

Simulator Screenshot - iPhone 15 Pro Max - 2024-04-10 at 09 29 56

### CODE REVIEW

File: lib/product_detail/widgets/product_detail_content.dart [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    final quantity =
+        context.select((ProductDetailCubit bloc) => bloc.state.quantity);
+
+    if (isLoading || product == null) {
+      return const Center(child: CircularProgressIndicator());
+    }
+
+    if (isFailure) {
+      return const Center(child: Text('Failed to load product'));
+    }
```
Comment (ncuillery) [inline lib/product_detail/widgets/product_detail_content.dart]: A bit of redundancy here 🙂 you have already taken care of those state in the parent widget (`ProductDetailView`)

File: lib/product_detail/widgets/product_detail_content.dart [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+    final galleryImages = product.galleryImages;
+    final galleryThumbnails = product.galleryThumbnails;
+
+    return SingleChildScrollView(
+      child: Column(
+        children: [
+          AspectRatio(
+            aspectRatio: galleryImages.isNotEmpty ? 1 : 1,
```
Comment (ncuillery) [inline lib/product_detail/widgets/product_detail_content.dart]: sooooo... `1`? 😄

File: lib/product_detail/widgets/product_detail_content.dart [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+                      onTap: () {
+                        setState(() {
+                          _selectedImage = index;
+                        });
+                      },
+                    ),
+                  );
+                },
+              ),
+            ),
```
Comment (ncuillery) [inline lib/product_detail/widgets/product_detail_content.dart:72]: I feel that this gallery (the zoomed photo and the thumbnails) could make a perfect widget for a eCom app accelerator

(can be done later)

**Review by ncuillery — APPROVED** [review]

File: lib/product_detail/widgets/product_detail_content.dart (hunk 3 above)
Comment (stephenrichter) [reply lib/product_detail/widgets/product_detail_content.dart:72]: Great point, I've opened a ticket for this https://github.com/BASE1com/flex-storefront/issues/16

**Pull request merged**